src/content/popup/KanjiEntry.tsx

```diff
@@ -0,0 +1,49 @@
+import { KanjiResult } from '@birchill/jpdict-idb';
+
+import { PopupOptions } from './popup';
+import { KanjiReferencesTable } from './KanjiReferencesTable';
+import { KanjiEntryTopPart } from './KanjiEntryTopPart';
+import { I18nProvider } from '../../common/i18n';
+
+type Props = {
+  entry: KanjiResult;
+  index: number;
+  options: PopupOptions;
+  selectState: 'unselected' | 'selected' | 'flash';
+};
+
+// temporary hack
+export function WrappedKanjiTable(props: Props) {
+  return (
+    <I18nProvider>
+      <KanjiEntry {...props} />
+    </I18nProvider>
+  );
+}
+
+export function KanjiEntry({ entry, index, options, selectState }: Props) {
+  // TODO: We may want to pull in the `classnames` package to simplify this kind of thing:
+  // https://www.npmjs.com/package/classnames.  It's a really common problem in React and
+  // somewhat annoying there's no built in support for class toggling sans string concatenation.
+  // Thankfully the classnames package is lightweight enough: https://bundlephobia.com/package/classnames@2.5.1
+  let tableClasses = 'kanji-table';
+  if (selectState === 'selected') {
+    tableClasses += ' -selected';
+  }
+  if (selectState === 'flash') {
+    tableClasses += ' -flash';
+  }
+
+  return (
+    <div class={tableClasses}>
+      <KanjiEntryTopPart entry={entry} index={index} options={options} />
+      {/* Reference row */}
+      {options.kanjiReferences && options.kanjiReferences.length && (
+        <KanjiReferencesTable
+          entry={entry}
+          kanjiReferences={options.kanjiReferences}
+        />
+      )}
```

Comment:
FWIW, I have very mixed feelings about this style as a way of doing conditional rendering, but I've acquiesced to it over time because it seems to be the accepted practice in the React world. Still, happy to change it depending on your feelings.

Reply:
Yeah, in general I think it's fine (as long as you ensure the value on the LHS is a bool, otherwise you sometimes yet `undefined` and things showing up in the output). In this case I think `!!options.kanjiReferences?.length && ...` would do the trick.